src/deploy/functions/release/sourceTokenScraper.ts

```diff
@@ -8,40 +13,54 @@ import { Timer } from "./timer";
  */
 export class SourceTokenScraper {
   private tokenValidDurationMs; // in ms
-  private tokenRefreshRequired = true;
-  private tokenRefreshTimer: Timer | undefined;
   private resolve!: (token?: string) => void;
   private promise: Promise<string | undefined>;
+  private expiration: bigint | undefined;
+  private tokenState: TokenState;
 
   constructor(validDurationMs = 1500000) {
     this.tokenValidDurationMs = validDurationMs;
     this.promise = new Promise((resolve) => (this.resolve = resolve));
+    this.tokenState = TokenState.NONE;
   }
 
-  // Token Promise will return undefined for the first caller
-  // (because we presume it's this function's source token we'll scrape)
-  // and then returns the promise generated from the first function's onCall
-  tokenPromise(): Promise<string | undefined> {
-    if (this.tokenRefreshRequired) {
-      this.tokenRefreshRequired = false;
-      this.tokenRefreshTimer = new Timer(this.tokenValidDurationMs);
-      return Promise.resolve(undefined);
+  async getToken(): Promise<string | undefined> {
+    if (this.tokenState === TokenState.NONE) {
+      this.tokenState = TokenState.INVALID;
+      return undefined;
+    } else if (this.tokenState === TokenState.INVALID) {
```

Comment:
I found that token states as described here bit confusing. `tokenState == INVALID` makes me think that token is invalid, but token doesn't actually exist at this point.

What about another name that closely maps to the state of the scraper, like `tokenFetchState` instead of `tokenState` and `FETCHING` and `INVALID`?